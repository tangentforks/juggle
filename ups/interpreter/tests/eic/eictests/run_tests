cg -I../../../include -oldc gethashwords.c
cg -I../../../include -oldc testAassert.c
cg -I../../../include -oldc testAbinary.c
cg -I../../../include -oldc testarray.c
cg -I../../../include -oldc testarray2.c
cg -I../../../include -oldc testarray3.c
cg -I../../../include -oldc testarray4.c
cg -I../../../include -oldc testaryptr.c
cg -I../../../include -oldc testassign.c
cg -I../../../include -oldc testcomma.c
cg -I../../../include -oldc testcompliant.c
cg -I../../../include -oldc testcond.c
cg -I../../../include -oldc testctype.c
cg -I../../../include -oldc testdeclerr.c
cg -I../../../include -oldc testenum.c
cg -I../../../include -oldc testerrno.c
cg -I../../../include -oldc testescape.c
cg -I../../../include -oldc testextern.c
cg -I../../../include -oldc testfloat.c
cg -I../../../include -oldc testflow.c
cg -I../../../include -oldc testfunc.c
cg -I../../../include -oldc testieee.c
cg -I../../../include -oldc testinit1.c
cg -I../../../include -oldc testinit2.c
cg -I../../../include -oldc testlexer.c
cg -I../../../include -oldc testlexerr.c
cg -I../../../include -oldc testlimits.c
cg -I../../../include -oldc testlimits2.c
cg -I../../../include -oldc testlogi.c
cg -I../../../include -oldc testparserr.c
cg -I../../../include -oldc testprintf.c
cg -I../../../include -oldc testqsort.c
cg -I../../../include -oldc testreturn.c
cg -I../../../include -oldc testscanf.c
cg -I../../../include -oldc testscop.c
cg -I../../../include -oldc testsign.c
cg -I../../../include -oldc testsizeof.c
cg -I../../../include -oldc teststring.c
cg -I../../../include -oldc teststru.c
cg -I../../../include -oldc teststru2.c
cg -I../../../include -oldc testtime.c
cg -I../../../include -oldc testtrace.c
cg -I../../../include -oldc testtypedef.c
cg -I../../../include -oldc testtypeerr.c
cg -I../../../include -oldc testtypes.c
cg -I../../../include -oldc testtypes2.c
cg -I../../../include -oldc testva.c
