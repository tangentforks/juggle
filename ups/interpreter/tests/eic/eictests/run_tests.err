gethashwords.c, 22: Function isalpha undefined
Assert should fail on line 10
testctype.c, 41: Function isalnum undefined
testctype.c, 45: Function isalpha undefined
testctype.c, 49: Function iscntrl undefined
testctype.c, 53: Function isdigit undefined
testctype.c, 57: Function isgraph undefined
testctype.c, 61: Function islower undefined
testctype.c, 65: Function isprint undefined
testctype.c, 69: Function ispunct undefined
testctype.c, 73: Function isspace undefined
testctype.c, 77: Function isupper undefined
testctype.c, 81: Function isxdigit undefined
testdeclerr.c, 10: syntax error
cg: Parse of testdeclerr.c failed
testfloat.c, 15: Function log10 undefined
testinit2.c, 132: Initialisation of local aggregate p not yet implemented
testinit2.c, 122: Initialisation of local aggregate y not yet implemented
testinit2.c, 109: Initialisation of local aggregate a not yet implemented
testinit2.c, 90: Initialisation of local aggregate un not yet implemented
testinit2.c, 56: Initialisation of local aggregate xs0 not yet implemented
testinit2.c, 64: Initialisation of local aggregate xs1 not yet implemented
testinit2.c, 72: Initialisation of local aggregate arg not yet implemented
testinit2.c, 29: Initialisation of local aggregate names not yet implemented
testinit2.c, 8: Initialisation of local aggregate a not yet implemented
testinit2.c, 9: Initialisation of local aggregate b not yet implemented
testlexerr.c, 8: Unterminated char constant
testlexerr.c, 8: syntax error
cg: Parse of testlexerr.c failed
testparserr.c, 13: syntax error
cg: Parse of testparserr.c failed
teststru2.c, 42: Scalar initialiser for aggregate variable `x'
teststru2.c, 53: Scalar initialiser for aggregate variable `x'
teststru2.c, 85: Scalar initialiser for aggregate variable `y'
teststru2.c, 86: Scalar initialiser for aggregate variable `x'
teststru2.c, 95: Scalar initialiser for aggregate variable `x'
cg: Parse of teststru2.c failed
testtypeerr.c, 20: syntax error
cg: Parse of testtypeerr.c failed
testtypes.c, 99: Sorry, ++ and -- on floating expressions not yet implemented
testtypes.c, 99: Sorry, ++ and -- on floating expressions not yet implemented
testtypes.c, 100: Sorry, ++ and -- on floating expressions not yet implemented
testtypes.c, 100: Sorry, ++ and -- on floating expressions not yet implemented
testtypes.c, 124: Type defaults to int
testtypes.c, 129: Type defaults to int
testtypes.c, 187: Undefined variable T8a
testtypes.c, 192: Undefined variable a
testtypes.c, 203: Warning: Cast ignored in initialisation of variable `PP'
	Expression: (pointer to void)(0)
	      type: pointer to void
cg: Parse of testtypes.c failed
error 13 hello 13.333 