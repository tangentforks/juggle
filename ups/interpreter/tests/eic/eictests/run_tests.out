assertion failed at testAassert.c(10)
0 0 0 0 0 0 :-> 0 0 0 0 0 0 
1 2 3 4 5 6 :-> 1 2 3 4 5 6 
1 2 0 4 5 0 :-> 1 2 0 4 5 0 
1 2 4 5 0 0 :-> 1 2 4 5 0 0 
14 :-> 14
[Hello, world!] :-> [Hello, world!]
5 :-> 5
Edmond :-> Edmond
Claire :-> Claire
Sarah :-> Sarah
Alexandra :-> Alexandra
Georgia :-> Georgia
1 2 3 4 5 0 :-> 1 2 3 4 5 0
1 0 2 0 3 0 :-> 1 0 2 0 3 0
1 2 3 4 5 6 7 0 :-> 1 2 3 4 5 6 7 0
abc :-> abc
120  :-> 120
450  :-> 450
1245 :-> 1245
2, 2, 5 :-> 2, 2, 5
::TooL0:00000::Me000:Yes00 :-> ::TooLo:00000::Me000:Yes00
ISO C compliant, line 11
ISO C compliant, line 20
shep = 4 :-> shep = 4
shep = 13 :-> shep = 13
colour = 0
RGB = 5
Ed = 1, CLAIRE = 2, SARAH = 1, ALEX 2, GEORGIA = 3
RED = 0,GREEN = 13,BLUE = 14 :-> RED = 0,GREEN = 13,BLUE = 14  
CHAR = 0,INT = 1,DOUBLE = 2 :-> CHAR = 0,INT = 1,DOUBLE = 2 
audible alert (bell) BEL  \a   7
backspace            BS   \b   8
horizontal tab       HT   \t   9
newline              LF   \n  10
vertical tab         VT   \v  11
formfeed             FF   \f  12
carriage return      CR   \r  13
double quote         "    \"  34
single quote         '    \'  39
question mark        ?    \?  63
backslash            \    \\  92
double getNumber( double f ):

    This function is passed a default value then requests a
    number from stdin.  If a RETURN is entered, the default
    value is returned.  If a new number is entered, the new
    number is returned. However, if an invalid number is
    entered, the function asks for a number again.

    This function is passed a default value then requests a
    number from stdin.  If a RETURN is entered, the default
    value is returned.  If a new number is entered, the new
    number is returned. However, if an invalid number is
    entered, the function asks for a number again.

    This function is passed a default value then requests a
    number from stdin.  If a RETURN is entered, the default
    value is returned.  If a new number is entered, the new
    number is returned. However, if an invalid number is
    entered, the function asks for a number again.


    This function is passed a default value then requests a
    number from stdin.  If a RETURN is entered, the default
    value is returned.  If a new number is entered, the new
    number is returned. However, if an invalid number is
    entered, the function asks for a number again.

    This function is passed a default value then requests a
    number from stdin.  If a RETURN is entered, the default
    value is returned.  If a new number is entered, the new
    number is returned. However, if an invalid number is
    entered, the function asks for a number again.

    This function is passed a default value then requests a
    number from stdin.  If a RETURN is entered, the default
    value is returned.  If a new number is entered, the new
    number is returned. However, if an invalid number is
    entered, the function asks for a number again.


012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678


012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678


-2147483648 = 2147483648.000000
assertion failed at testlimits.c(40)
-128

---------------test1
|       365|%|10x|       16D|       555|
|      +365|+365      |+000000365|      -365|
|     0x16d|0X16D     |%|10s|%|.10s|
|000000000000000000013|%!21.2f|
|000000000000001000.00|%!21.2f|
---------- test2
0.000000 :-> 0.000000
1.000000 :-> 1.000000
10.000000 :-> 10.000000
12.345000 :-> 12.345000
1234567.800000 :-> 1234567.800000
1000.00 :-> 1000.00
999.99 :-> 999.99
0.100000 :-> 0.100000
1.000000 :-> 1.000000
0.000000e+00 :-> 0.000000e+00
1.000000e+00 :-> 1.000000e+00
1.000000e+01 :-> 1.000000e+01
1.234568e+06 :-> 1.234568e+06
1.234568E+06 :-> 1.234568E+06
4.94066e-300 :-> 4.94066e-300
-0.00000 :-> -0.00000
-1.234568e+11 :-> -1.234568e+11
-4.94066e-300 :-> -4.94066e-300
-123456789999.399994 :-> -123456789999.400000
-0.00000 :-> -0.00000
---------- test3
0 :-> 0
13 :-> 13
0 :-> 0
1 :-> 1
0e+00 :-> 0e+00
1e+11 :-> 1e+11
2e+11 :-> 2e+11
-0 :-> -0
-123456790000 :-> -123456790000
-1e+11 :-> -1e+11
-5e-300 :-> -5e-300
---------- test4a
0 :-> 0
10 :-> 10
0.02 :-> 0.02
1.23457e+06 :-> 1.234578e+06
123.456 :-> 123.456
1E+06 :-> 1E+06
---------- test4b
0 :-> 0e+00
1e+01 :-> 1e+01
0.02 :-> 0
1e+06 :-> 1e+06
-2E+02 :-> -2E+02
1e+06 :-> 1e+06
---------- test4c
0. :-> 0.0e+00
1.e+01 :-> 1.0e+01
0.02 :-> 0.0
1.e+06 :-> 1.2e+06
+2.E+02 :-> +1.5E+02
1.e+06 :-> 1.0e+06

---------------test5
|--------------------|
|%!*g|
|%!*.*f|
|%!*.*f|
|%!*.*f|
|%!*.*f|
|%!*g|
|%!*G|
|%!*g|
|%#!*g|
|%!*.*f|
|%!*.*f|
|%!*.*f|
|%!*.*f|
|%!*g|
|%!*G|
|%#!*.*g|
|%+!*.*f|
|%+!*.*f|
|%+!*.*f|
|%+!*.*f|
|--------------------|
<fini>

110  86  22 100  85  74  25 109  95 116 
 22  25  74  85  86  95 100 109 110 116 
 22 = 22
 116 = 116
 95 = 95
This file should generate 2 dangling pointer warnings
Okay 1
Okay 1
count = [3]  quant = [2] units = [quarts] item = [oil]
count = [2]  quant = [-12.8] units = [degrees] item = []
count = [0]  quant = [-12.8] units = [] item = []
count = [3]  quant = [10] units = [LBS] item = [dirt]
count = [3]  quant = [100] units = [rgs] item = [energy]
	--- EiC test signed values ---
char c; short s; int i; long l;  float f; double d;

c = -6 s = 0 i = 5 l = -100 f = 2.600000 d = 0.900000 :->
c = -6 s = 0 i = 5 l = -100 f = 2.6 d = 0.9

-c  = 6 :-> -c = 6
~s  = -1 :-> ~s = -1
f * c = -15.6  :-> f * c = -15.6
f / d = 2.888889  :-> f / d = 2.88889
l / f = -38.461540  :-> l / f = -38.4615
i << 2 = 20 :-> i << 2 = 20
	--- EiC test unsigned values ---
unsigned char uc; short unsigned us; unsigned ui; unsigned long ul;

uc = 0 us = 0  ui = 0  ul = 0:->
uc = 0 us = 0  ui = 0 ul = 0

125 :-> 1
2
5

:-> error 13 hello 13.333
error 13 hello 13.333 
6 34 -4 
-5 6 98 23 4 
